### PR TITLE
Compiler warning part 4

### PR DESCRIPTION
**Pull request by RincewindsHat**
Parts of #1848
Easier to digest :-)

### CODE REVIEW

File: plugins/check_curl.c [hunk 1]
```diff
@@ -1133,6 +1133,11 @@ redir (curlhelp_write_curlbuf* header_buf)
     &status_line.http_major, &status_line.http_minor, &status_line.http_code, &status_line.msg, &msglen,
     headers, &nof_headers, 0);
 
+	if (res == -1) {
```
Comment (andreasbaumann) [inline plugins/check_curl.c:1140]: Why only here? There is also one in 2304, 2362?
phr_parse_response is not returning -1 on error. What should be done on -2 (partial response)?

**Review by andreasbaumann — COMMENTED** [review]
This needs a closer look IMHO..

**Review by andreasbaumann — COMMENTED** [review]
Do I see this correctly, this is also a refactoring taking out the max_fd function for future reuse in other places (which is a good thing)?

**Comment by RincewindsHat** [comment]
Implicitely yes, because it was previously included in places where it was not used.

File: plugins/check_curl.c (hunk 1 above)
Comment (RincewindsHat) [reply plugins/check_curl.c:1140]: I don't know :-| What would you suggest? Die in all non-OK cases?

**Comment by RincewindsHat** [comment]
Cold and outdated

**Pull request closed without merging**